internal/dataplane/parser/translate_httproute.go

```diff
@@ -305,40 +298,79 @@ func generateKongRouteFromHTTPRouteMatches(
 	// stripPath needs to be disabled by default to be conformant with the Gateway API
 	r.StripPath = kong.Bool(false)
 
-	// attach the plugins to be applied to the given route
-	if len(plugins) != 0 {
-		if r.Plugins == nil {
-			r.Plugins = make([]kong.Plugin, 0, len(plugins))
+	seenMethods := make(map[string]struct{})
+
+	var hasRedirectFilter bool
+	for _, f := range filters {
+		if f.Type == gatewayv1beta1.HTTPRouteFilterRequestRedirect {
+			hasRedirectFilter = true
 		}
-		r.Plugins = append(r.Plugins, plugins...)
 	}
 
-	seenMethods := make(map[string]struct{})
-
 	for _, match := range matches {
```

Comment:
This patch makes this for loop substantially bigger, WDYT about extracting this to a separate `func`?

Reply:
extracted, let's see if I broke something 😁